internal/app/management/run.go

```diff
@@ -0,0 +1,456 @@
+	}
+	for _, secret := range secrets.Items {
+		if err := secretsClient.Delete(secret.Name, nil); err != nil {
+			return nil, fmt.Errorf("error deleting secret %s", secret.Name)
+		}
+	}
+
+	var deleteResponse = &module.ModuleDeleteResponse{
+		Name: r.Name,
+	}
+
+	return deleteResponse, nil
+}
+
+func (s *server) List(ctx context.Context, r *module.ModuleListRequest) (*module.ModuleListResponse, error) {
+	// Find deployments with matching label
+	deploymentsClient := k.client.AppsV1().Deployments(k.namespace)
+	deployments, err := deploymentsClient.List(metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("createdBy=%s", mgmt.ID),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error listing deployments with label %s", mgmt.ID)
+	}
+	var list = &module.ModuleListResponse{}
+	for _, deployment := range deployments.Items {
+		list.Names = append(list.Names, deployment.Name)
+	}
+	return list, nil
+}
+
+func (s *server) Get(ctx context.Context, r *module.ModuleGetRequest) (*module.ModuleGetResponse, error) {
+	// Get the deployment with the given name
+	deploymentsClient := k.client.AppsV1().Deployments(k.namespace)
+	deployments, err := deploymentsClient.Get(r.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("error getting deployments with name %s", r.Name)
+	}
+	var getResponse = &module.ModuleGetResponse{
+		Name: deployments.Name,
+	}
+	return getResponse, nil
+}
+
+func encodeBase64(s string) string {
+	const lineLen = 70
+	encLen := base64.StdEncoding.EncodedLen(len(s))
+	lines := encLen/lineLen + 1
+	buf := make([]byte, encLen*2+lines)
+	in := buf[0:encLen]
+	out := buf[encLen:]
+	base64.StdEncoding.Encode(in, []byte(s))
+	k := 0
+	for i := 0; i < len(in); i += lineLen {
+		j := i + lineLen
+		if j > len(in) {
+			j = len(in)
+		}
+		k += copy(out[k:], in[i:j])
+		if lines > 1 {
+			out[k] = '\n'
+			k++
+		}
+	}
+	return string(out[:k])
+}
+
+func homeDir() string {
+	if h := os.Getenv("HOME"); h != "" {
+		return h
+	}
+	return os.Getenv("USERPROFILE") // windows
+}
+
+func getClientSet() (*kubernetes.Clientset, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.WithError(err).Warn("failed getting in-cluster config attempting to use kubeconfig from homedir")
+		var kubeconfig string
+		if home := homeDir(); home != "" {
+			kubeconfig = filepath.Join(home, ".kube", "config")
+		}
+
+		if _, err := os.Stat(kubeconfig); os.IsNotExist(err) {
+			log.WithError(err).Panic("kubeconfig not found in homedir")
+		}
+
+		// use the current context in kubeconfig
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+		if err != nil {
+			log.WithError(err).Panic("getting kubeconf from current context")
+			return nil, err
+		}
+	}
+
+	// create the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.WithError(err).Error("Getting clientset from config")
+		return nil, err
+	}
+
+	return clientset, nil
+}
+
+func int32Ptr(i int32) *int32 { return &i }
```

Comment:
Nit: Azure autorest has a "to" package which gives lots of these utility function. https://github.com/lawrencegripper/ion/blob/master/internal/app/dispatcher/providers/kubernetes.go#L249